statistics/cmsketch.go

```diff
@@ -75,12 +174,18 @@ func (c *CMSketch) queryValue(sc *stmtctx.StatementContext, val types.Datum) (ui
 }
 
 // QueryBytes is used to query the count of specified bytes.
-func (c *CMSketch) QueryBytes(bytes []byte) uint32 {
-	h1, h2 := murmur3.Sum128(bytes)
+func (c *CMSketch) QueryBytes(d []byte) uint32 {
+	if c.topnindex != nil {
+		if cnt, ok := c.topnindex[hack.String(d)]; ok {
```

Comment:
Still not efficient. It takes a lot to compare strings for every query.

Reply:
How about:

```
func (c *CMSketch) QueryBytes(d []byte) uint32 {
	h1, h2 := murmur3.Sum128(d)
	if c.topnindex != nil {
		cnt, ok := c.topnindex[h1]
		if ok {
			for k := range cnt {
				if cnt[k].h2 == h2 && bytes.Equal(d, cnt[k].data) {
					return cnt[k].count
				}
			}
		}
	}
	return c.queryHashValue(h1, h2)
}
```

Reply:
Yes, I think it is better.